test/flash_test/src/flash_test.c

```diff
@@ -170,11 +179,139 @@ flash_cli_cmd(int argc, char **argv)
     return -1;
 }
 
+
+/*
+ * Returns # of ops done within 2 seconds.
+ */
+int
+flash_speed_test(int flash_dev, uint32_t addr, int sz, int move)
+{
+    int rc;
+    int cnt = 0;
+    int off = 0;
+    int start_time;
+    int end_time;
+    void *data_buf;
+
+    data_buf = malloc(sz);
+    if (!data_buf) {
+        return -1;
+    }
+
+    /*
+     * Catch start of a tick.
+     */
+    start_time = os_time_get();
+    while (1) {
+        end_time = os_time_get();
+        if (end_time != start_time) {
+            start_time = end_time;
+            break;
+        }
+    }
+
+    /*
+     * Measure for 2 secs.
+     */
+    do {
+        rc = hal_flash_read(flash_dev, addr + off, data_buf, sz);
+        if (rc) {
+            console_printf("hal_flash_read(%d, 0x%x, %d) = %d\n",
+              flash_dev, (unsigned int)addr + off, (unsigned int)sz, rc);
+            return -1;
+        }
+        assert(rc == 0);
```

Comment:
Since there is error handling, the `assert` doesn't seem necessary here!

Reply:
True